src/core/isEmpty.js

```diff
@@ -1,8 +1,14 @@
 /** @license ISC License (c) copyright 2016 original and current authors */
 /** @author Ian Hofmann-Hicks (evil) */
 const isObject = require('./isObject')
+const isMonoid = require('./isMonoid')
+const equals = require('./equals')
 
 function isEmpty(x) {
+  if(isMonoid(x)) {
+    return equals(x, x.constructor.empty())
```

Comment:
Sorry, I totally forgot about fantasy land name prefixing,
One last change and then I think this will be GTG.

So this should be something like:
```js
const fl = require('../flNames')

return equals(x, (x.constructor[fl.empty] || x.empty)())
```

Reply:
Actually to account for pre 1.x fl it should probably be:
```js
return equals(x, (x.constructor[fl.empty] || x.constructor.empty || x.empty)()
```